src/requirements.ts

```diff
@@ -145,3 +170,39 @@ function invalidJavaHome(reject, cause: string) {
     });
   }
 }
+
+export async function installManagedJre() {
+  return vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: 'Downloading JRE' },
+    async (progress, cancelToken) => {
+      const options = await PlatformInformation.GetPlatformInformation();
+      console.log(`Detected options ${JSON.stringify(options)}`);
+      return jre.download({
```

Comment:
You are mixing async/await then/catch pattern in the same function here I would rather chose one or the other.